ledger/appcow_test.go

```diff
@@ -1299,3 +1330,24 @@ func TestCowDelKey(t *testing.T) {
 	a.Panics(func() { c.DelKey(getRandomAddress(a), aidx, false, key, 0) })
 	a.Panics(func() { c.DelKey(addr, aidx+1, false, key, 0) })
 }
+func TestCowAppendLog(t *testing.T) {
+	a := require.New(t)
+
+	addr := getRandomAddress(a)
+	aidx := basics.AppIndex(1)
+	c := getCow([]modsData{
+		{addr, basics.CreatableIndex(aidx), basics.AppCreatable},
+	})
+
+	val := strings.Repeat("val", 100)
+	tv := basics.TealValue{Type: basics.TealBytesType, Bytes: val}
+	err := c.AppendLog(aidx, tv)
+	a.Error(err)
+	a.Contains(err.Error(), "value too long")
```

Comment:
Why should this case fail? Since `val := strings.Repeat("val", 100)`, `val` should be 300 bytes long, which is well under the 1000 byte limit -- let me know if I'm missing something here.

Reply:
this is because I had `len(value.Bytes) > cb.proto.MaxAppBytesValueLen` in AppendLog. it should be compared to MaxLogSize instead.